### ISSUE TITLE
Error when run in Docker for Windows with mounted volume.

### ISSUE THREAD

**Issue by MicahZoltu** [Level/levelup#393]
Follow-up from comments here: https://github.com/Level/levelup/issues/222#issuecomment-228646063

Note: I am a user of a product that uses LevelDB, so my repro case unfortunately isn't very refined.

Repro steps:
1. Windows 10
2. Docker for Windows beta `1.12.0-rc2`
3. Mount a drive ![image](https://cloud.githubusercontent.com/assets/886059/16538151/a8a03cd8-3fcf-11e6-843c-da780966e9b0.png)
4. Create a folder somewhere on the mounted (e.g., `d:\foo`)
5. `docker run --rm -it --name storj-farmer -v //d/foo:/data oreandawe/storj-farmer:latest setup`
6. Answer questions with defaults except `Enter the path to store configuration and data` and `Enter the path to store your encrypted private key`
7. `Enter the path to store configuration and data`: `/data/bar`
8. `Enter the path to store your encrypted private key`: `/data/id_ecdsa`
9. `docker run --rm -v //d/foo:/data oreandawe/storj-farmer:latest start --datadir /data/bar --password <password_chosen_during_setup>`

This should result in the following error:

```
events.js:141
      throw er; // Unhandled 'error' event
      ^
OpenError: IO error: /data/foo/farmer.db: Invalid argument
    at /usr/lib/node_modules/storjshare-cli/node_modules/storj/node_modules/levelup/lib/levelup.js:119:34
    at /usr/lib/node_modules/storjshare-cli/node_modules/storj/node_modules/leveldown/node_modules/abstract-leveldown/abstract-leveldown.js:39:16
```

It is entirely possible that this is a problem with Storj but LevelDB isn't giving a terrible useful error here.  At the least, LevelDB should give a more useful error indicating what exactly went wrong (e.g., couldn't open file, file doesn't exist, couldn't write to file, etc.).  At the moment, I'm not even sure what the problem is since `/data/foo/farmer.db` is a folder on-disk (not a file).

It is also possible that this is a problem with LevelDB on Docker for Windows Beta.

**Comment by coreybutler** [comment]
I know this is an old thread now, but did you experiment with using a non-mounted directory for your data (i.e. add in the Dockerfile and embed in the image)? Docker still has some issues with volume mounting and being able to identify the difference between a file and a directory. This is especially true with non-physical disks (i.e. the share).

**Comment by Zoltu** [comment]
@coreybutler Everything works great when the files are written/read all within the container (no mounted volume).  However, in this case there is a hard requirement on persisting the on-disk database across container runs so that solution won't work here.

**Comment by coreybutler** [comment]
@Zoltu - I was afraid of that. If it works when added to the image but not when mounted as a volume, then you're running into a Docker issue... not a levelup issue. There's not much levelup can do when the disk i/o isn't working, even from an error reporting perspective.

Docker hasn't worked very well with shares on unix file systems, let alone the Windows shares. I've tried to do something similar with Docker+Windows Shares with a 100% failure rate. However; the latest 1.12 (full release, not RC) has substantial updates and volumes got an overhaul. Perhaps an upgrade would fix the issue.

Bottom line - I'm kind of guessing at solutions, but I am confident the problem is not with levelup/leveldb.
